### ISSUE TITLE
Searching for something that doesn't exist causes search function to fail

### ISSUE THREAD

**Issue by rdickey** [spinnaker/spinnaker#2561]
*Title*
Searching for something that doesn't exist causes search function to fail

*Cloud Provider*
AWS - Reproduced with AWS, but cloud provider may not matter.

*Environment*
Running custom AMI of Spinnaker on EC2:

```
dpkg -l | grep spinnaker
ii  spinnaker                            0.83.0                            all          spinnaker
ii  spinnaker-clouddriver                2.3.0                             all          spinnaker-clouddriver
ii  spinnaker-deck                       2.1182.0                          all          spinnaker-deck
ii  spinnaker-echo                       123.456.789                       all          spinnaker-echo
ii  spinnaker-fiat                       0.40.0                            all          spinnaker-fiat
ii  spinnaker-front50                    1.121.2                           all          spinnaker-front50
ii  spinnaker-gate                       4.21.2                            all          spinnaker-gate
ii  spinnaker-igor                       2.0.0                             all          spinnaker-igor
ii  spinnaker-orca                       6.20.0                            all          spinnaker-orca
ii  spinnaker-rosco                      0.104.0                           all          spinnaker-rosco
```

*Feature Area*
UI - Search - Search

Description
From the Search page, if you search for an something that does not exist (or make a typo in your search that leads to zero results) the browser will get stuck on "Loading". If you have DevTools open, you will see an error:

```
TypeError: Cannot read property '0' of undefined
    at InfrastructureSearchServiceV2.search (app.js:177165)
    at t.project (app.js:65727)
    at t._next (vendor.bundle.js:31)
    at t.next (vendor.bundle.js:13)
    at t.next (vendor.bundle.js:13)
    at InfrastructureSearcher.query (app.js:65737)
    at app.js:177510
    at f.$digest (vendor.bundle.js:41)
    at f.$apply (vendor.bundle.js:41)
    at vendor.bundle.js:41 undefined
```

*Steps to Reproduce*

Click the top-level Search page
Search for something that doesn't exist, for example: foodoesntexist
Observe the "Loading". If DevTools are open, the above error is displayed.  Further searches, even on existing applications/clusters, will only show "Loading".

*Additional Details*
This is not as high-impact as #2555 as it does not throw exceptions in a tight loop or freeze the browser.  A simple reloading of the page returns to its original functionality

Quick troubleshooting suggests this may have been caused by, or related to, this commit in deck v2.1177.0: https://github.com/spinnaker/deck/commit/0d2fbf846cb12a8d38bdae3ccdfb91d1d4a3ecbf#diff-994f1e8f611b63e2a5572283a1f4da2fR28
We have confirmed the bug is not present in deck version 2.1168, but haven't tested other versions.

**Comment by sbwsg** [comment]
This should now be resolved in the latest tagged version of deck.  Please reopen if you continue to see the errant behaviour.  Thanks for the issue report!

**Comment by rdickey** [comment]
Thanks; I have confirmed that this is fixed in 2.1188.0
